### PR TITLE
rambox: unmaintain & mark as insecure

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change
Rambox hasn't had a stable release in a while and an increasing number
of issues which is why I don't intend to use this anymore.

While taking a closer look at the source I also realized that it uses
Electron 7.2.4[1]. This is not only EOLed[2], it also contains a few
security vulnerabilities which is why I decided to mark it as insecure.

A few (most likely not all) vulnerabilities can be found by looking at
the Electron 7 changelog[3]: after 7.2.4 there were a few more releases
with security backports - mostly from Chromium. Security issues that
were found later on (and are probably exploitable on the dependency
chain of rambox) aren't listed here. I only added two issues that seemed
applicable to `rambox`, but I haven't researched enough to check the
other ones.

[1] https://github.com/ramboxapp/community-edition/blob/0.7.7/package.json#L70
[2] https://www.electronjs.org/docs/tutorial/support#currently-supported-versions
[3] https://www.electronjs.org/releases/stable?version=7

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](./CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mathnerd314** [comment]
There is some work to update the Electron version (https://github.com/ramboxapp/community-edition/pull/2897), but it's not merged because it breaks Skype (https://github.com/ramboxapp/community-edition/issues/2947). I guess marking the vulnerabilities is the best Nixpkgs can do, unless the Skype issue is fixed or an active fork appears.

**Comment by Ma27** [comment]
Yes, I agree, that's also one of the reasons why I decided against removing the package altogether. However there's no action from the upstream maintainers (don't know about the state of rambox-pro though), so marking it as insecure for now is fine.

But since I'm tired of an application that crashes whenever I try to download a file, that won't change my decision to step down as maintainer :)

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 1]
```diff
@@ -21,8 +21,11 @@ in mkRambox rec {
     description = "Free and Open Source messaging and emailing app that combines common web applications into one";
     homepage = "https://rambox.pro";
     license = licenses.mit;
-    maintainers = with maintainers; [ ma27 ];
+    maintainers = with maintainers; [ ];
     platforms = ["i686-linux" "x86_64-linux"];
     hydraPlatforms = [];
+    knownVulnerabilities = [
+      "Electron 7.2.4 is EOL and contains at least the following vulnerabilities: CVE-2020-6458, CVE-2020-6460 and more (https://www.electronjs.org/releases/stable?version=7)"
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:28]:
```suggestion
      "Electron 7.2.4 is EOL and contains at least the following vulnerabilities: CVE-2020-6458, CVE-2020-6460 and more (https://www.electronjs.org/releases/stable?version=7). Consider switching to ferdi"
```

Comment (Ma27) [reply pkgs/applications/networking/instant-messengers/rambox/default.nix]: Done as it's the best alternative I'm aware of. Though I'm a little tempted to state that messenger aggregation is an unsolved CS problem after having used ferdi for a while %)

Comment (Mathnerd314) [reply pkgs/applications/networking/instant-messengers/rambox/default.nix]: I guess Android kind of solves it, in that if you want to read lots of messages you can setup all the clients as apps on Android phone(s) and sync the notifications to your desktop with KDE Connect. And then you'd set up screen-sharing for replying. But buying a phone is probably not anyone's first choice of solution.

**Pull request merged**